src/workers/parser/mapBindings.js

```diff
@@ -49,10 +126,6 @@ function replaceNode(ancestors, node) {
   }
 }
 
-function hasDestructuring(node) {
-  return node.declarations.some(declaration => t.isPattern(declaration.id));
```

Comment:
As it is, this logic will still fail with array patterns, so if we do end up going with this pattern, we'll still want to bail out here for array patterns, just not object patterns.

Reply:
Support for array destructuring has been added.